magicctapipe/scripts/lst1_magic/semi_automatic_scripts/merging_runs.py

```diff
@@ -0,0 +1,362 @@
+"""
+This script splits the proton MC data sample into "train"
+and "test", deletes possible MC failed runs (only those files
+that end up with a size < 1 kB), and generates the bash
+scripts to merge MC and real data files by calling the script "merge_hdf_files.py"
+in the following order:
+
+MAGIC:
+
+Merge the subruns into runs for M1 and M2 individually.
+
+MC:
+
+Merges all MC runs in a node
+
+Usage:
+$ merging_runs (-c config.yaml) (-t analysis_type)
+
+If you want to merge only the MAGIC or only the MC data,
+you can do as follows:
+
+Only MAGIC:
+$ merging_runs -t onlyMAGIC (-c config.yaml)
+
+Only MC:
+$ merging_runs -t onlyMC (-c config.yaml)
+"""
+
+import argparse
+import glob
+import logging
+import os
+from pathlib import Path
+
+import joblib
+import numpy as np
+import yaml
+from tqdm import tqdm
+
+from magicctapipe import __version__
+from magicctapipe.scripts.lst1_magic.semi_automatic_scripts.clusters import (
+    rc_lines,
+    slurm_lines,
+)
+
+__all__ = ["cleaning", "split_train_test", "merge", "mergeMC"]
+
+logger = logging.getLogger(__name__)
+logger.addHandler(logging.StreamHandler())
+logger.setLevel(logging.INFO)
+
+
+def cleaning(list_of_nodes):
+
+    """
+    This function looks for failed runs in each MC node and remove them.
+
+    Parameters
+    ----------
+    list_of_nodes : array of str
+        List of nodes where the function will look for failed runs.
+    """
+
+    cwd = os.getcwd()
+    for i in tqdm(range(len(list_of_nodes)), desc="Cleaning failed runs"):
+        os.chdir(list_of_nodes[i])
+        os.system('find . -type f -name "dl1_[gphe]*_zd*_az*.h5" -size -1k -delete')
```

Comment:
we were discussing that automatic removal is dangerous and we should rather understand the reason for files failing at earlier step, and @Elisa-Visentin wrote that she removed (commented out) this, but I still see that the removal is being done. Was is not commented out after or, or it turned out to be needed for some other reason?

Reply:
I removed it from real data. I can remove it also from MC and add a warning in the readme (i.e.,' if you are processing MCs, check Dl1s for failed jobs and remove broken files before merging') Broken files must be removed not to make the merging jobs fail, but if you process MCs you must know what you are doing (and know the pipeline!) so you can take care of manually removing files

Reply:
since the MC part will be removed, there is no point in modifying this part now.
I am resolving this comment

Reply:
BTW, this part was removed in commit 3c47546 a couple of weeks ago (then probably this change got lost in the 'Outdated' comments)